app/src/main/java/com/github/se/studybuddies/viewModels/SharedTimerViewModel.kt

```diff
@@ -1,64 +1,55 @@
 package com.github.se.studybuddies.viewModels
 
 import android.os.CountDownTimer
-import android.util.Log
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
-import com.github.se.studybuddies.data.TimerState
 import com.github.se.studybuddies.database.DatabaseConnection
-import com.google.firebase.database.DataSnapshot
-import com.google.firebase.database.DatabaseError
-import com.google.firebase.database.ValueEventListener
 import kotlinx.coroutines.*
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 
-class SharedTimerViewModel private constructor(private val groupUID: String) : ViewModel() {
+class SharedTimerViewModel(private val groupUID: String) : ViewModel() {
   private val databaseConnection = DatabaseConnection()
-  private val _timerValue = MutableStateFlow(0L) // Holds the current timer value in milliseconds
+  private val _timerValue = MutableStateFlow(0L)
   val timerValue: StateFlow<Long> = _timerValue
 
-  private val _timerEnd = MutableStateFlow(false) // Indicates whether the timer has ended
+  private val _timerEnd = MutableStateFlow(false)
   val timerEnd: StateFlow<Boolean> = _timerEnd
 
   private var isRunning = false
   private var countDownTimer: CountDownTimer? = null
 
-  companion object {
-    private val instances = mutableMapOf<String, SharedTimerViewModel>()
-
-    fun getInstance(groupUID: String): SharedTimerViewModel {
-
-      return instances.getOrPut(groupUID) { SharedTimerViewModel(groupUID) }
+  init {
+    viewModelScope.launch {
+      syncTimerWithFirebase()
+      startPeriodicSync()
     }
   }
 
-  init {
-    subscribeToTimerUpdates()
+  private fun startPeriodicSync() {
+    viewModelScope.launch {
+      while (isActive) {
+        syncTimerWithFirebase()
+        delay(2000) // Appeler toutes les 2 secondes
+      }
+    }
   }
```

Comment:
It's not the more elegant way. Look at PR #217 where with the changes, it detect automatically the update and doesn't need to call it every 2s

Reply:
I will address this issue in another pr where I correct a lot of non good   practice of coding

Reply:
there will be another pr that address this issues